app/scripts/lib/experiments/base.js

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+  An experiment interface that helps organize A/B experiments.
+  It helps keep the state, organize testing groups and log experiment metrics.
+ */
+
+define([
+  'underscore',
+  'lib/storage',
+  'lib/url'
+],
+function (_, Storage, Url) {
+  'use strict';
+
+  var FORCE_GROUP_TYPE = 'forceExperimentGroup';
+  var storage = Storage.factory('localStorage');
+
+  function BaseExperiment() {
+    // nothing to do.
+  }
+
+  _.extend(BaseExperiment.prototype, {}, {
+    /**
+     * Name of the experiment. Used in metrics and view logic.
+     */
+    _name: null,
+
+    /**
+     * Experiment notifications. Used to notify an experiment about something that happened in the view
+     */
+    notifications: {},
+
+    /**
+     * The A/B testing group type, see https://en.wikipedia.org/wiki/A/B_testing for details.
+     */
+    _groupType: null,
+
+    /**
+     * Initialize the experiment
+     *
+     * @returns {boolean}
+     *  Returns 'true' when the experiment successfully initialized. Otherwise 'false'.
+     */
+    initialize: function (name, options) {
+      this._initialized = false;
+      // all experiments require these options
+      if (! name || ! options || ! options.window || ! options.metrics || ! options.able || ! options.user || ! options.notifications) {
```

Comment:
Could also use DeMorgan's theorem to turn this from a series of `|| !` to a `!` and a bunch of `&&`'s